src/routers/__tests__/TabRouter-test.js

```diff
@@ -600,4 +602,60 @@ describe('TabRouter', () => {
       ],
     });
   });
+
+  test('Handles the tab selection action when moving to tabrouter', () => {
```

Comment:
I don't fully understand the test name. What is "moving to tabrouter"?

In the test, you use StackRouter and TabRouter. I'd like to ask you to rename the test so that it is clear:
What is being tested
Given what circumstances
What is the expected result

It doesn't have to come in this order, but we have the luxury of being able to use a full sentence to describe the test case and I'd like the sentence to answer these questions. Thank you for the PR! 👍

Reply:
How does "Can navigate to a selected tab after resetting to the tabrouter" sound?

Reply:
I think I'd like it to be even more specific. I'd go for something like "Given a StackRouter with nested TabRouter, it is possible to reset the StackRouter and navigate to an arbitrary tab (`BarTab`) in the TabRouter by supplying sub-actions to the RESET action."

Reply:
that works for me! i'll change it to that.